frontend/src/metabase/parameters/components/ValuesSourceModal/ValuesSourceModal.unit.spec.tsx

```diff
@@ -0,0 +1,210 @@
+      ).toBeInTheDocument();
+    });
+
+    it("should show unique non-null mapped fields values", async () => {
+      setup({
+        parameter: createMockUiParameter({
+          fields: [
+            new Field(createMockField({ id: 1 })),
+            new Field(createMockField({ id: 2 })),
+          ],
+        }),
+        fieldsValues: [
+          createMockFieldValues({
+            field_id: 1,
+            values: [["A"], [null], ["B"], ["A"]],
+          }),
+          createMockFieldValues({
+            field_id: 2,
+            values: [["B", "Remapped"], ["C"]],
+          }),
+        ],
+      });
+
+      await waitFor(() => {
+        expect(screen.getByRole("textbox")).toHaveValue("A\nB\nC");
+      });
+    });
+  });
+
+  describe("card source", () => {
+    it("should show a message when there are no text columns", async () => {
+      setup({
+        parameter: createMockUiParameter({
+          values_source_type: "card",
+          values_source_config: {
+            card_id: 1,
+          },
+        }),
+        cards: [
+          createMockCard({
+            id: 1,
+            name: "Products",
+          }),
+        ],
+      });
+
+      expect(
+        await screen.findByText(/This question doesn’t have any text columns/),
+      ).toBeInTheDocument();
+    });
+
+    it("should allow to select only text fields", async () => {
+      const { onSubmit } = setup({
+        parameter: createMockUiParameter({
+          values_source_type: "card",
+          values_source_config: {
+            card_id: 1,
+          },
+        }),
+        cards: [
+          createMockCard({
+            id: 1,
+            name: "Products",
+            result_metadata: [
+              createMockField({
+                id: 1,
+                display_name: "ID",
+                base_type: "type/BigInteger",
+                semantic_type: "type/PK",
+              }),
+              createMockField({
+                id: 2,
+                display_name: "Category",
+                base_type: "type/Text",
+                semantic_type: "type/Category",
+              }),
+            ],
+          }),
+        ],
+      });
+
+      userEvent.click(
+        await screen.findByRole("button", { name: /Pick a column/ }),
+      );
+      expect(
+        screen.queryByRole("heading", { name: "ID" }),
+      ).not.toBeInTheDocument();
+
+      userEvent.click(screen.getByRole("heading", { name: "Category" }));
+      userEvent.click(screen.getByRole("button", { name: "Done" }));
+      expect(onSubmit).toHaveBeenCalledWith("card", {
+        card_id: 1,
+        value_field: ["field", 2, null],
+      });
+    });
+  });
+
+  describe("list source", () => {
+    it("should set static list values", () => {
+      const { onSubmit } = setup();
+
+      userEvent.click(screen.getByRole("radio", { name: "Custom list" }));
+      userEvent.type(screen.getByRole("textbox"), "Gadget\nWidget");
```

Comment:
I know the code already handles trimming spaces, but should we assert that behavior in this test too? Like maybe spamming a few more new lines or appending or prepending spaces to an option.